app/services/unpublish_service.rb

```diff
@@ -5,9 +5,7 @@ def withdraw(edition, public_explanation, user = nil)
     Document.transaction(requires_new: true) do
       edition.document.lock!
       check_unpublishable(edition)
-
-
-      withdrawal = Withdrawal.new(public_explanation: public_explanation)
+      withdrawal = new_withdrawal(edition, public_explanation)
```

Comment:
minor but since we don't use .new on Withdrawal anymore it might be clearer to call this create_withdrawal